### ISSUE TITLE
Question: Can I use webgui with NAS (DARTS) and lightning evaluator ? 

### ISSUE THREAD

**Issue by mashu** [microsoft/nni#5278]
I would like to access NNI webgui, but when I start the model as follows

```
    # Load dataset
    data = pd.read_csv("data/ERR.tsv", sep='\t', nrows=8192)
    training_data = data.sample(frac=0.90, random_state=25)
    testing_data = data.drop(training_data.index)
    train_dataset = MyDataset(training_data, "aux.tsv")
    test_dataset = MyDataset(testing_data, "aux.tsv")
    train_dataloader = nni.trace(DataLoader)(train_dataset, batch_size=64, collate_fn=PadSequence())
    test_dataloader = nni.trace(DataLoader)(test_dataset, batch_size=64, collate_fn=PadSequence())
    
    lightning = pl.Lightning(LightNet(),
                         pl.Trainer(max_epochs=500, accelerator='gpu'),
                         train_dataloaders=pl.DataLoader(train_dataset, batch_size=64, collate_fn=PadSequence(),num_workers=6),
                         val_dataloaders=pl.DataLoader(test_dataset, batch_size=64, collate_fn=PadSequence(), num_workers=6))

    exploration_strategy = strategy.DARTS()
    exp_config = RetiariiExeConfig('local')
    exp_config.experiment_name = 'mymodel'
    exp_config.trial_concurrency = 12
    exp_config.max_trial_number = 2048
    exp_config.trial_gpu_number = 1
    exp_config.training_service.use_active_gpu = True
    export_formatter = 'dict'
    exp_config.execution_engine = 'oneshot'
    
    experiment = RetiariiExperiment(Net(), lightning, [], exploration_strategy)
    experiment.run(exp_config, 8081)
```
It runs in terminal
```
GPU available: True (cuda), used: True
TPU available: False, using: 0 TPU cores
IPU available: False, using: 0 IPUs
HPU available: False, using: 0 HPUs
LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0]

  | Name  | Type     | Params
-----------------------------------
0 | model | LightNet | 1.4 M 
-----------------------------------
1.4 M     Trainable params
0         Non-trainable params
1.4 M     Total params
5.592     Total estimated model params size (MB)
Training: 0it [00:00, ?it/s]...
```
but is not starting experiment webgui which is requested by providing 8081 port to run(...) call on the experiment object.

**Question**: Is this expected? For random strategy I can see what is going on within webgui but for oneshot DART, I can't?
Why do I need to provide port for run(...) function if it's not opening that port?

**Environment**:
- NNI version: 2.19
- Training service (local|remote|pai|aml|etc): local
- Client OS: GNU/Linux Debian
- Python version: 3.10.8
- PyTorchversion: 2.13
- Is conda/virtualenv/venv used?: Conda
- Is running in Docker?: No

**Comment by matluster** [comment]
The short answer is NO. You can't use NNI's WebUI to visualize one-shot strategy. You can use [tensorboard](https://nni.readthedocs.io/en/stable/tutorials/darts.html#launch-experiment) though.

> Why do I need to provide port for run(...) function if it's not opening that port?

For historical reasons. One-shot was added later at around NNI v2.7 or v2.8. Before that, `run()` only serves multi-trial strategies.

**Comment by matluster** [comment]
I thought your question is about visualization. You can use any strategy for NAS.

And I didn't understand this part.

> I would like to optimize to have both small number of parameters and good accuracy, but DARTs is not using accuracy but gradients right?

**Comment by Lijiaoa** [comment]
@mashu Could you update your issue for debug?

**Comment by mashu** [comment]
I think this issue can be closed, it was answered, that it is not possible to use standard webgui with oneshot strategies.
Yes, this issue was about visualization, ignore secondary question.
